src/ResponseFactory.php

```diff
@@ -101,10 +101,9 @@ private function addAccessControl(RequestInterface $request, ?array $accessToken
         // If the search request doesnt have a query, we add a default match_all query.
         $query = $body->query ?? ['match_all' => new stdClass()];
 
-        // Create the access control filter
         $filter = new stdClass();
         $filter->terms = [
-            AccessTokenMappingHelper::ACCESS_TOKEN_PROPERTY => $accessTokens ?: ['anonymous'],
+            AccessTokenMappingHelper::ACCESS_TOKEN_PROPERTY . '.keyword' => $accessTokens ?: ['anonymous'],
```

Comment:
Are we using this constant somewhere else? If not we should add `.keyword` to it directly.

Reply:
yes we are, using it to exclude _access_tokens from source as in here:

$body->_source = [
            'excludes' => [AccessTokenMappingHelper::ACCESS_TOKEN_PROPERTY],
        ];

and a few other places. I think concatenating here is fine as this is the only place where we are filtering